episodes/preparing_for_machine_learning.md

```diff
@@ -0,0 +1,438 @@
+
+![](fig/t1_vT3_large.jpg){alt='T1 v T3'}
+
+*Sourced from [Bernd L. Schmitz, Andrik J. Aschoff, Martin H.K. Hoffmann and Georg Grön, Advantages and Pitfalls in 3T MR Brain Imaging: A Pictorial Review ,American Journal of Neuroradiology October 2005, 26 (9) 2229-2237](https://www.ajnr.org/content/26/9/2229)*
+
+Different contrast levels make the caudate and thalami far more or less apparent. The radiomics of these brain images are different in terms of contrast, and probably some other parameters, even though it is literally the same patient.
+
+We can't simply build a dataset based on unharmonized derived data. On the other hand, we may not have access to the images, to even guess how things are going differently.
+
+We suggest the following approach: 
+
+1. Compare the data yourself in terms of descriptive statistics and what you know about the two patient groups (do you expect the same counts in each group? why or why not?).
+2. Consider a harmonization package.
+
+Below are a three examples from the world of neuroimaging:
+
+|             | `neurocombat`    | `haca3`        | `autocombat`           |
+|-------------|------------------|----------------|------------------------|
+| License     | MIT for Python/ Artistic for R  | Missing?   | Apache 2.0  |
+| Last updated| 2021             | 2024           |2022        | 
+| Programming language | Python or R     | Python      | Python         | 
+| Organ/area and modality | brain MRI     | brain MRI              | brain MRI  |
+| Data type |  derived values    | images        | derived values       | 
+| Notes     | no release yet standard   | versioned but not released   | no releases not versions           |
+| Original Website     | [GitHub](https://github.com/Jfortin1/ComBatHarmonization) | [GitHub](https://github.com/lianruizuo/haca3) | [GitHub](https://github.com/Alxaline/ComScan)    | 
+| Early Publication |[doi: 10.1016/j.neuroimage.2017.08.047](https://doi.org/10.1016/j.neuroimage.2017.08.047)| [doi:10.1016/j.compmedimag.2023.102285](https://doi.org/10.1016/j.compmedimag.2023.102285)  |[doi: 10.1038/s41598-022-16609-1 ](https://doi.org/10.1038/s41598-022-16609-1)|
+| Verioned website |  [versioned on cvasl](https://github.com/brainspinner/cvasl/tree/main/cvasl/vendor/neurocombat) | [versioned on GitHub](https://github.com/lianruizuo/haca3)| [versioned on cvasl](https://github.com/brainspinner/cvasl/tree/main/cvasl/vendor/autocombat)  
```

Comment:
The [link](https://github.com/brainspinner/cvasl/tree/main/cvasl/vendor/autocombat) to the cvals versioning gives: 404 - page not found. Please fix it :)

Reply:
Ah, yes, oops, fixed